R/Methods-FAKinSumResults.R

```diff
@@ -147,16 +149,45 @@ setMethod("result", "FAKinSumResults", function(object, method="BH"){
         affected_id=affIds,
         family=fams,
         affected=rep(length(resList$affected), length(resList$sumKinship)),
+        ksgrp=NA,
         kinship_sum=resList$sumKinship,
         freq=resList$frequencyKinship,
         pvalue=resList$pvalueKinship,
         padj=p.adjust(resList$pvalueKinship, method=method),
         stringsAsFactors=FALSE)
     rownames(res) <- as.character(res$affected_id)
-    return(res[order(res$pvalue, -res$kinship_sum), , drop=FALSE])
+    res <- res[order(res$pvalue, -res$kinship_sum), , drop=FALSE]
+    padj <- res$padj
+    names(padj) <- rownames(res)
+    res$ksgrp <- define.result.groups.ks(kinship(object), padj, th=cutoff)
+    return(res)
 })
 
-
+##' Define groups of related affected individuals for KS test.
+##' @param kin Kinship matrix.
+##' @param padj Named vector with adjusted P values of all affected individuals.
+##' @param th  Threshold of `padj` for inclusion of affected individuals.
+##' @return A named vector of group numbers or NAs, the names correspond to the
+##' IDs of the affected individuals. An NA indicates that the affected
+##' individual was above the threshold. The named vector corresponds to the
+##' entries in vector `padj`.
+define.result.groups.ks <- function(kin, padj, th=0.05)
```

Comment:
should we maybe also add an additional parameter `minKinship = 0` to the function (and then also to `doShareKinship`) that allows to define which individuals to group? I guess we would not like to put e.g. 3rd cousins into the same "kin" group as their kin is only very low. The default could be still `0` but we could then also use a different kinship threshold - as I would like to not put, like said, third grade cousins into the same kinship group.

Reply:
Sure, that's easy.

Reply:
Excellent! Please re-request my review once you've pushed to this PR and are OK with me reviewing it.